filebeat/crawler/prospector.go

```diff
@@ -97,35 +100,89 @@ func (p *Prospector) Run(wg *sync.WaitGroup) {
 				return
 			case event := <-p.harvesterChan:
```

Comment:
Prospector level filtering. But as you mentioned in the chat, this can perhaps better be done via some interfaces.